src/rpc/blockchain.cpp

```diff
@@ -316,6 +324,7 @@ UniValue getblockheader(const UniValue& params, bool fHelp)
             "  \"confirmations\" : n,   (numeric) The number of confirmations, or -1 if the block is not on the main chain\n"
             "  \"height\" : n,          (numeric) The block height or index\n"
             "  \"version\" : n,         (numeric) The block version\n"
+            + strprintf("  \"versionHex\" : \"00000000\", (string) The block version formatted in hexadecimal when version is greater than %d\n", VERSIONBITS_LAST_OLD_BLOCK_VERSION) +
```

Comment:
Kind of. It's a little more generic than that though. That threshold is the highest version number that I know for sure will be treated as a simple integer. I wanted to avoid redundancy for the easy to ascertain version numbers that are definitely treated as integers.

I was trying to avoid being too specific to BIP9 because it allows for two future mechanisms (top bits 010 and 011). I thought of using `>= VERSIONBITS_TOP_BITS`, but that ignores [5, 0x1fffffff] which I think could be re-purposed too. I don't see any rule limiting that range's use to only being simple integers.

I should probably just keep it simple and always display `versionHex`. My redundancy limiting kinda feels like unnecessary optimization.